rules-java/api/src/main/java/org/jboss/windup/rules/apps/mavenize/MavenizationService.java

```diff
@@ -0,0 +1,384 @@
+            projectModel.setProjectType(suffix); // FIXME: Remove when WINDUP-983 is fixed.
+            return suffix;
+        }
+
+        // TODO: Should we try something more? Used APIs? What if it's a source?
+
+        return "unknown";
+    }
+
+
+    String deriveAppropriateArtifactId(ProjectModel projectModel)
+    {
+        String resultName = null;
+        String name = projectModel.getName();
+        name: {
+            if (name == null)
+                break name;
+            if (name.length() > 40)
+                break name;
+            if (name.contains(" "))
+                break name;
+            resultName = name;
+        }
+
+        if (resultName == null)
+        {
+            resultName = projectModel.getRootFileModel().getFileName();
+        }
+
+        resultName = removeVersion(resultName);
+        resultName = normalizeDirName(resultName);
+
+        if (resultName == null)
+            resultName = "unknownName-" + RandomStringUtils.randomAlphanumeric(4);
+
+        // See WINDUP-1015
+        if (resultName.length() > 40)
+            resultName = resultName.substring(0,40) + "-" + RandomStringUtils.randomAlphanumeric(4);
+
+
+        return resultName;
+    }
+
+
+    /**
+     * Remove 1.0.0 from foo-1.0.0.jar
+     */
+    String removeVersion(String resultName)
+    {
+        if(resultName == null)
+            return null;
+
+        // Regex test at http://fiddle.re/dnwhca
+        return resultName.replaceFirst("[-_]\\d+(\\.\\d+)+(-SNAPSHOT|[.-_](?i:CR|RC|GA|Alpha|Beta|b|milestone|m|Final|RELEASE)[.-_]?\\d+|-incubating)?(?=[-_.]\\w+)", "");
+    }
+
+
+    /**
+     * Context of the mavenization - things to carry around.
+     */
+    static class MavenizationContext {
+        private Path mavenizedBaseDir;
+        private Pom rootPom;
+        private Set<Pom> knownSubmodules = new HashSet<>();
+        private String unifiedVersion;
+        private String unifiedGroupId;
+        private String unifiedAppName;
+        private Pom bom; // BOM shared by all other submodules.
+        private GraphContext graphContext;
+
+
+        public Path getMavenizedBaseDir() {
+            return mavenizedBaseDir;
+        }
+
+        public Pom getRootPom() {
+            return rootPom;
+        }
+
+        public Set<Pom> getKnownSubmodules() {
+            return knownSubmodules;
+        }
+
+        public String getUnifiedVersion() {
+            return unifiedVersion;
+        }
+
+        public String getUnifiedGroupId() {
+            return unifiedGroupId;
+        }
+
+        public String getUnifiedAppName() {
+            return unifiedAppName;
+        }
+
+        public Pom getBom() {
+            return bom;
+        }
+
+
+        public GraphContext getGraphContext() {
+            return graphContext;
+        }
+
+    }
+
+    /**
+     * This will be used to determine which BOM to take.
+     * Currently we only have these tags: eap, eap7.
+     */
```

Comment:
So this code already takes in mind the different java ee version. So i think we need to parametrize the BOMs specifically to this (javaee-6 ->BOM and javaee-7 -> BOM as I commented in BOM template)